test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -70,6 +70,121 @@ TEST(NoProxyTest, EmptyEntries) {
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
 }
 
+// Test entries with CIDR blocks (Class A) in 'no_proxy' list.
+TEST(NoProxyTest, CIDRClassAEntries) {
+  ScopedSetEnv no_proxy("foo.com,192.168.0.255/8");
+  auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "http://proxy.google.com");
+  EXPECT_EQ(HttpProxyMapper().MapName("dns:///192.0.1.1:443", &args),
+            absl::nullopt);
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), absl::nullopt);
+
+  EXPECT_EQ(HttpProxyMapper().MapName("dns:///193.0.1.1:443", &args),
+            "proxy.google.com");
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER),
+            "193.0.1.1:443");
+}
+
+// Test entries with CIDR blocks (Class B) in 'no_proxy' list.
+TEST(NoProxyTest, CIDRClassBEntries) {
+  ScopedSetEnv no_proxy("foo.com,192.168.0.255/16");
+  auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "http://proxy.google.com");
+  absl::optional<std::string> name_to_resolve;
```

Comment:
I suspect this showed up since you probably copied the template of `EmptyEntries` but it does not seem to be needed.